spec/integration/outputs/update_spec.rb

```diff
@@ -88,6 +88,21 @@ def get_es_output( options={} )
       insist { r["_source"]["counter"] } == 4
     end
 
+    it "should increment a counter with event/doc 'count' variable with event/doc as upsert and inline script" do
+      subject = get_es_output({
+        'document_id' => "123",
+        'doc_as_upsert' => true,
+        'script' => 'ctx._source.counter += event["count"]',
+        'script_lang' => 'groovy',
+        'script_type' => 'inline'
+      })
+      subject.register
+      subject.receive(LogStash::Event.new("count" => 3 ))
+      subject.flush
+      r = @es.get(:index => 'logstash-update', :type => 'logs', :id => "123", :refresh => true)
```

Comment:
@salimane, you already have a document_id '123' with a field "counter" = 1 at start of the test. 
we wants to check the upsert here, not the update ? you could change the document_id to '456' instead.

But with this script I think it will failed because in this case ctx._source.counter doesn't exist and groovy will fail on '+='